GitFyle.Core.Api.Tests.Unit/Controllers/Contributions/ContributionsControllerTests.Exceptions.Delete.cs

```diff
@@ -0,0 +1,213 @@
+﻿// ----------------------------------------------------------------------------------
+// Copyright (c) The Standard Organization: A coalition of the Good-Hearted Engineers
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using GitFyle.Core.Api.Models.Foundations.Contributions;
+using GitFyle.Core.Api.Models.Foundations.Contributions.Exceptions;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using RESTFulSense.Clients.Extensions;
+using RESTFulSense.Models;
+using Xeptions;
+
+namespace GitFyle.Core.Api.Tests.Unit.Controllers.Contributions
+{
+    public partial class ContributionsControllerTests
+    {
+        [Theory]
+        [MemberData(nameof(ValidationExceptions))]
+        public async Task ShouldReturnBadRequestOnDeleteIfValidationErrorOccursAsync(Xeption validationException)
+        {
+            // given
+            Guid someId = Guid.NewGuid();
+
+            BadRequestObjectResult expectedBadRequestObjectResult =
+                BadRequest(validationException.InnerException);
+
+            var expectedActionResult =
+                new ActionResult<Contribution>(expectedBadRequestObjectResult);
+
+            this.contributionServiceMock.Setup(service =>
+                service.RemoveContributionByIdAsync(It.IsAny<Guid>()))
+                    .ThrowsAsync(validationException);
+
+            // when
+            ActionResult<Contribution> actualActionResult =
+                await this.contributionsController.DeleteContributionByIdAsync(someId);
+
+            // then
+            actualActionResult.ShouldBeEquivalentTo(expectedActionResult);
+
+            this.contributionServiceMock.Verify(service =>
+                service.RemoveContributionByIdAsync(It.IsAny<Guid>()),
+                    Times.Once);
+
+            this.contributionServiceMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldReturnFailedDependencyOnDeleteIfReferenceErrorOccursAsync()
+        {
+            // given
+            Guid someId = Guid.NewGuid();
```

Comment:
```suggestion
            Guid someContributionId = Guid.NewGuid();
```
also fix across the other tests